File: ganilib-core/src/main/java/com/gani/lib/select/FragmentItemSelect.java
package com.gani.lib.select;

import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TabHost;

import com.gani.lib.R;
import com.gani.lib.screen.GActivity;
import com.gani.lib.screen.GFragment;
import com.gani.lib.ui.layout.GLinearLayout;

import org.jetbrains.annotations.NotNull;

public abstract class FragmentItemSelect<I extends SelectableItem, T extends SelectGroup.Tab> extends GFragment {
//  static final String PARAM_SELECTED_ITEMS = "selectedItems";
//  static final String PARAM_MULTISELECT = "multiselect";
//
  public static final String RETURN_ITEMS = "items";

  private SelectGroup.ViewHelper<T> tabHelper;

  private T selectedTab;
//  private ControllerItemSelect<I, T> controller;

  protected FragmentItemSelect(SelectGroup.ViewHelper<T> helper) {
    this.tabHelper = helper;
  }

//  protected abstract SelectGroup getDefaultGroup();

  private View onCreateView(LayoutInflater inflater, int layoutId, Bundle savedInstanceState) {
    View fragmentLayout = inflater.inflate(layoutId, null);

    selectedTab = tabHelper.initView(savedInstanceState, getActivity(),
        (TabHost) fragmentLayout.findViewById(R.id.tabhost),
        inflater.inflate(R.layout.tabcontent_common_list, null));
    initTabsIn(inflater);

    return fragmentLayout;
  }

//  @Override
//  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
////    View fragmentLayout = inflater.inflate(R.layout.fragment_common_list, null);
////
////    selectedTab = tabHelper.initView(savedInstanceState, getActivity(),
////        (TabHost) fragmentLayout.findViewById(R.id.tabhost),
////        inflater.inflate(R.layout.tabcontent_common_list, null));
////    initTabsIn(inflater);
////
////    return fragmentLayout;
//    return onCreateView(inflater, R.layout.fragment_common_list, savedInstanceState);
//  }

  @Override
  public void initContent(@NotNull GActivity activity, @NotNull GLinearLayout container) {
//    super.initContent(activity, container);

    container.append(LayoutInflater.from(getContext()).inflate(R.layout.fragment_common_list,null));

//    View.inflate(getContext(), )
  }

  @Override
  public void onActivityCreated(Bundle savedInstanceState) {
    super.onActivityCreated(savedInstanceState);

    if (savedInstanceState == null) {
      onRefresh();
    }
  }

  @Override
  protected void onRefresh() {
    super.onRefresh();

    // TODO
//    controller.asyncRetrieveDataIfNecessary();
  }

  @Override
  public void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    tabHelper.save(outState, selectedTab);
  }

  @Override
  public void onDestroy() {
    // TODO
//    controller.onDestroy();
    super.onDestroy();
  }

  private void initTabsIn(LayoutInflater inflater) {
    RecyclerView listView = (RecyclerView) tabHelper.findView(R.id.list_common);

    // TODO
//    controller = createController(listView, selectedTab, getArguments().getBoolean(PARAM_MULTISELECT));
//
//    getLoaderManager().initLoader(0, null, controller);
//
    tabHelper.initTabs(new TabContentSwitcher(), selectedTab);

    initList(listView);
  }

  protected abstract void initList(RecyclerView listView);

//  protected abstract ControllerItemSelect<I, T> createController(RecyclerView listView, T group, boolean multiSelect);



  private class TabContentSwitcher implements TabHost.OnTabChangeListener {
    @Override
    public void onTabChanged(String topicTabName) {
      selectedTab = tabHelper.getTab(topicTabName);

      // TODO
//      controller.setSelectGroup(selectedTab);
//
//      controller.asyncRetrieveDataIfNecessary();
    }
  }
}

File: glib-core/src/main/java/com/glib/core/ui/select/FragmentItemSelect.kt
package com.gani.lib.select

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TabHost
import android.widget.TabHost.OnTabChangeListener
import androidx.recyclerview.widget.RecyclerView
import com.glib.core.R
import com.glib.core.screen.GActivity
import com.glib.core.screen.GFragment
import com.glib.core.screen.GScreenContainer
import com.glib.core.ui.list.GRecyclerView
import com.glib.core.ui.select.SelectableItem
import com.glib.core.ui.view.ViewHelper

abstract class FragmentItemSelect<I : SelectableItem?> protected constructor(
//    helper: ViewHelper?
) : GFragment() {
//    private val tabHelper: ViewHelper?
//    private var selectedTab: T? = null

//    private fun onCreateView(
//        inflater: LayoutInflater?,
//        layoutId: Int,
//        savedInstanceState: Bundle?
//    ): View? {
//        val fragmentLayout = inflater!!.inflate(layoutId, null)
////        selectedTab = tabHelper.initView(
////            savedInstanceState, getActivity(),
////            fragmentLayout!!.findViewById<View?>(R.id.tabhost) as TabHost?,
////            inflater.inflate(R.layout.tabcontent_common_list, null)
////        )
////        initTabsIn(inflater)
//
//        val listView: RecyclerView = fragmentLayout.findViewById(R.id.list_common) as RecyclerView
//        initList(listView)
//        return fragmentLayout
//    }

    override fun initContent(activity: GActivity, container: GScreenContainer) {
//        val layout = LayoutInflater.from(getContext()).inflate(R.layout.fragment_common_list, null)
        val listView = GRecyclerView(activity)
        listView.width(ViewGroup.LayoutParams.MATCH_PARENT)
        container.append(listView)

//        val listView = layout.findViewById(R.id.list_common) as RecyclerView
        initList(listView)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (savedInstanceState == null) {
            onRefresh()
        }
    }

//    override fun onRefresh() {
//        super.onRefresh()
//    }

//    override fun onSaveInstanceState(outState: Bundle) {
//        super.onSaveInstanceState(outState)
//        tabHelper.save(outState, selectedTab)
//    }

//    fun onDestroy() {
//        super.onDestroy()
//    }

//    private fun initTabsIn(inflater: LayoutInflater?) {
//        val listView: RecyclerView = tabHelper.findView(R.id.list_common) as RecyclerView
//
////        tabHelper.initTabs(TabContentSwitcher(), selectedTab)
//        initList(listView)
//    }

    protected abstract fun initList(listView: RecyclerView)

//    private inner class TabContentSwitcher : OnTabChangeListener {
//        override fun onTabChanged(topicTabName: String?) {
//            selectedTab = tabHelper.getTab(topicTabName)
//        }
//    }

    companion object {
        val RETURN_ITEMS = "items"
    }

//    init {
//        tabHelper = helper
//    }
}